### PR TITLE
aws-secrets-controller: basic logging for secret reading time

### LINKED ISSUE #27933: additional logging to verify async timeouts around `librdkafka` calls in the Kafka sink
## Problem

During investigation of https://github.com/MaterializeInc/incidents-and-escalations/issues/33 one instance was observed where the sink seemingly got permanently stuck and needed a manual restart. This suggests an async call did not time out, but there is insufficient logging to be sure.

## Solution

Add additional logging to confirm if an async call in this area fails to time out.

## See also

* https://github.com/MaterializeInc/incidents-and-escalations/issues/33
* https://app.slack.com/client/TL68GSMGF/C04DJT084KA
* https://app.slack.com/client/TL68GSMGF/C06CQC8LW4V

### PR DESCRIPTION
**Pull request by benesch**
We have some evidence that reading secrets from AWS might stall indefinitely. Add a basic log line that indicates when an AWS secret read starts and ends to prove or disprove this hypothesis if it occurs again in production.

We log at an `info!` level so that we have visibility into every secrets read. These are relatively infrequent, so this shouldn't be a meaningful increase in log volume. We don't want to log at a lower level because the hang has only occured rarely, so we want the logs on by default for when it happens next.

Fix #27933.

### Motivation

  * This PR fixes a recognized bug.

### Tips for reviewer

None.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

File: src/aws-secrets-controller/src/lib.rs [hunk 1]
```diff
@@ -209,15 +211,22 @@ impl AwsSecretsClient {
...
-            .client
-            .get_secret_value()
-            .secret_id(self.secret_name(id))
-            .send()
-            .await?
-            .secret_binary()
-            .ok_or_else(|| anyhow!("internal error: secret missing secret_binary field"))?
-            .to_owned()
-            .into_inner())
+        info!(%id, "reading secret from AWS");
```
Comment (danhhz) [inline src/aws-secrets-controller/src/lib.rs:214]: Do you want to stick a unique id (e.g. a UUID) in in case two happen concurrently?

**Review by danhhz — APPROVED** [review]

File: src/aws-secrets-controller/src/lib.rs (hunk 1 above)
Comment (benesch) [reply src/aws-secrets-controller/src/lib.rs:214]: Ooh, good thinking.

Comment (sdht0) [reply src/aws-secrets-controller/src/lib.rs:214]: > in in case

case in point 😄

**Review by alex-hunt-materialize — APPROVED** [review]
+1 to Dan's UUID suggestion.

File: src/aws-secrets-controller/src/lib.rs (hunk 1 above)
Comment (benesch) [reply src/aws-secrets-controller/src/lib.rs]: Good idea. Done!

**Pull request merged**